src/views/virtualmachines/details/tabs/configuration/storage/components/tables/disk/utils/getEditDiskStates.ts

```diff
@@ -46,40 +41,32 @@ export const getEditDiskStates: UseEditDiskStates = (vm, diskName, vmi) => {
     const volume = volumes?.find(({ name }) => name === diskName);
     // volume consists of 2 keys:
     // name and one of: containerDisk/cloudInitNoCloud
-    const volumeSource = Object.keys(volume).find((key) =>
+    const volumeSource = Object.keys(volume).find((key: VolumeTypes) =>
       [
-        volumeTypes.CONTAINER_DISK,
-        volumeTypes.DATA_VOLUME,
-        volumeTypes.PERSISTENT_VOLUME_CLAIM,
+        VolumeTypes.CONTAINER_DISK,
+        VolumeTypes.DATA_VOLUME,
+        VolumeTypes.PERSISTENT_VOLUME_CLAIM,
       ].includes(key),
     );
 
-    const isBootDisk = diskName === getBootDisk(vm)?.name;
+    if (!isEmpty(disk)) {
+      draftState.diskInterface = getDiskInterface(disk);
+      draftState.diskType = diskTypes[getDiskDrive(disk)];
+      draftState.sharable = disk.shareable;
+      draftState.lunReservation = disk.lun?.reservation;
+    }
 
-    if (volumeSource === sourceTypes.EPHEMERAL) {
-      initialDiskSourceState.ephemeralSource = volume.containerDisk?.image;
-      return { diskSize: DYNAMIC, diskSource: sourceTypes.EPHEMERAL, isBootDisk };
+    if (volumeSource === SourceTypes.EPHEMERAL) {
+      draftState.containerDisk = { url: volume.containerDisk?.image };
+      draftState.diskSource = SourceTypes.EPHEMERAL;
+      draftState.diskSize = DYNAMIC;
```

Comment:
do we have those code chunks in a few files above?

Reply:
yes I have changed here to have some common functions